go/libraries/doltcore/schema/typeinfo/linestring.go

```diff
@@ -0,0 +1,180 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package typeinfo
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/dolthub/go-mysql-server/sql"
+
+	"github.com/dolthub/dolt/go/store/types"
+)
+
+// This is a dolt implementation of the MySQL type Point, thus most of the functionality
+// within is directly reliant on the go-mysql-server implementation.
+type linestringType struct {
+	sqlLinestringType sql.LinestringType
+}
+
+var _ TypeInfo = (*linestringType)(nil)
+
+var LinestringType = &linestringType{sql.LinestringType{}}
+
+// ConvertTypesLinestringToSQLLinestring basically makes a deep copy of sql.Linestring
+func ConvertTypesLinestringToSQLLinestring(l types.Linestring) sql.Linestring {
+	points := make([]sql.Point, len(l.Points))
+	for i, p := range l.Points {
+		points[i] = ConvertTypesPointToSQLPoint(p)
+	}
+	return sql.Linestring{SRID: l.SRID, Points: points}
+}
+
+// ConvertNomsValueToValue implements TypeInfo interface.
+func (ti *linestringType) ConvertNomsValueToValue(v types.Value) (interface{}, error) {
+	// Expect a types.Linestring, return a sql.Linestring
+	if val, ok := v.(types.Linestring); ok {
+		return ConvertTypesLinestringToSQLLinestring(val), nil
+	}
+	// Check for null
+	if _, ok := v.(types.Null); ok || v == nil {
+		return nil, nil
+	}
+	return nil, fmt.Errorf(`"%v" cannot convert NomsKind "%v" to a value`, ti.String(), v.Kind())
+}
+
+// ReadFrom reads a go value from a noms types.CodecReader directly
+func (ti *linestringType) ReadFrom(nbf *types.NomsBinFormat, reader types.CodecReader) (interface{}, error) {
+	k := reader.ReadKind()
+	switch k {
+	case types.LinestringKind:
+		l, err := reader.ReadLinestring()
+		if err != nil {
+			return nil, err
+		}
+		return ti.ConvertNomsValueToValue(l)
+	case types.NullKind:
+		return nil, nil
+	}
+
+	return nil, fmt.Errorf(`"%v" cannot convert NomsKind "%v" to a value`, ti.String(), k)
+}
+
+func ConvertSQLLinestringToTypesLinestring(l sql.Linestring) types.Linestring {
+	points := make([]types.Point, len(l.Points))
+	for i, p := range l.Points {
+		points[i] = ConvertSQLPointToTypesPoint(p)
+	}
+	return types.Linestring{SRID: l.SRID, Points: points}
+}
+
+// ConvertValueToNomsValue implements TypeInfo interface.
+func (ti *linestringType) ConvertValueToNomsValue(ctx context.Context, vrw types.ValueReadWriter, v interface{}) (types.Value, error) {
+	// Check for null
+	if v == nil {
+		return types.NullValue, nil
+	}
+
+	// Convert to sql.LinestringType
+	line, err := ti.sqlLinestringType.Convert(v)
+	if err != nil {
+		return nil, err
+	}
+
+	return ConvertSQLLinestringToTypesLinestring(line.(sql.Linestring)), nil
+}
+
+// Equals implements TypeInfo interface.
+func (ti *linestringType) Equals(other TypeInfo) bool {
+	if other == nil {
+		return false
+	}
+	if ti2, ok := other.(*linestringType); ok {
+		return ti.sqlLinestringType.Type() == ti2.sqlLinestringType.Type()
```

Comment:
This may technically work, but generally we're comparing the parameters assuming the `TypeInfo`s line up. If there are specific parameters that need to be checked, then we should compare those directly (see `blobstring.go`). If there aren't any parameters to check, then the `TypeInfo` check itself is fine (see `time.go`).